x-pack/plugins/synthetics/server/common/mem_writable.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from '@kbn/logging';
+import archiver from 'archiver';
+import { Writable, WritableOptions } from 'node:stream';
+
+export class MemWritable extends Writable {
+  private _buffer: Buffer;
+  constructor(opts?: WritableOptions) {
+    super(opts);
+    this._buffer = Buffer.from('');
+  }
+
+  public get buffer(): Buffer {
+    return this._buffer;
+  }
+
+  _write(
+    chunk: any,
+    _encoding: BufferEncoding,
+    callback: (error?: Error | null | undefined) => void
+  ): void {
+    try {
+      this._buffer = Buffer.concat([this._buffer, chunk]);
+      callback();
+    } catch (e) {
+      callback(e);
+    }
+  }
+}
+
+function wrapInlineInProject(inlineJourney: string) {
+  return `import { journey, step, expect } from '@elastic/synthetics';
+
+journey('inline', ({ page, context, browser, params, request }) => {
+  ${inlineJourney}
+});
+`;
+}
+
+export async function inlineToProjectZip(
+  inlineJourney: string,
+  monitorId: string,
+  logger: Logger
+): Promise<string> {
+  const mWriter = new MemWritable();
+  try {
+    await new Promise((resolve, reject) => {
+      const archive = archiver('zip', {
+        zlib: { level: 9 },
+      });
+      archive.on('error', reject);
+      mWriter.on('close', resolve);
+      archive.pipe(mWriter);
+      archive.append(wrapInlineInProject(inlineJourney), {
+        name: 'inline.journey.ts',
+      });
+      archive.finalize();
+    });
+  } catch (e) {
+    logger.error(`Failed to create zip for inline monitor ${monitorId}`);
```

Comment:
instead of catching and rethrowing. It would be better for this func to return promise and reject with log/error. Avoids try/catch flow.

Reply:
[9de233d](https://github.com/elastic/kibana/pull/176094/commits/9de233d712a0cc7532dcb93aa17ca7d5dcab38eb)